patroni/ha.py

```diff
@@ -25,6 +26,7 @@ def __init__(self, patroni):
         self.old_cluster = None
         self.recovering = False
         self._async_executor = AsyncExecutor()
+        self._disable_sync = False
```

Comment:
> Instead touch_member needs to be protected by a mutex to avoid getting the value overwritten by stale one.

That's actually what I meant.

> it would be better to add a timeout for while_not_sync_standby (2*loop_wait should be enough).

Timeout is also important. But I think it still make sense to "serialize" touch_member calls...

> * Standby publishes nosync
> * Standby checks sync state and thinks it can shut down.

There are 2 seconds between these actions, so chances are very low...